service/src/SSTIO.cpp

```diff
@@ -243,6 +243,9 @@ namespace geopm
 
             for (auto &batch : m_mbox_read_batch) {
                 int err = m_ioctl->mbox(batch.get());
+                if (err == -1 && errno == EBUSY) {
+                    err = m_ioctl->mbox(batch.get());
```

Comment:
```suggestion
                if (err == -1 && errno == EBUSY) {
                    errno = 0;
                    err = m_ioctl->mbox(batch.get());
```

Reply:
This change should be made throughout.  If errno is handled, it should be reset.